Tests/AppTests/SearchShowModelTests.swift

```diff
@@ -143,5 +144,13 @@ class SearchShowModelTests: XCTestCase {
 
         XCTAssertEqual(matchingKeywords, ["keyword2", "keyword4"])
     }
+    
+    func test_SearchShow_Model_canonicalURLAllowList() throws {
```

Comment:
We needed to write an `app.test` test here as this situation only occurs with query parameters not included in `API.SearchController.Query `. Instead, these situations would come from incoming links or someone just typing into a browser address bar.

This seemed like a reasonable way to test it, would you agree @finestructure?

Reply:
Actually, there's a slightly better way to get the snapshot and even render it in a more readable way:

```swift
    @MainActor
    func test_SearchShow_Model_canonicalURLAllowList() async throws {
        let req = Vapor.Request(application: app,
                                url: "search?query=alamo&page=2&utm_campaign=test&utm_source=email",
                                on: app.eventLoopGroup.next())
        let html = try await SearchController.show(req: req)
        assertSnapshot(matching: html.render(indentedBy: .spaces(2)), as: .html)
    }
```

→

```
<!DOCTYPE html>
<html lang="en"><!--Version: test--><!--DB Id: db-id-->
  <head>
    <meta name="robots" content="noindex"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
    <meta charset="UTF-8"/>
    <meta property="og:site_name" content="The Swift Package Index"/>
    <link rel="canonical" href="http://localhost:8080/search?query=alamo&page=2"/>
    <meta name="twitter:url" content="http://localhost:8080/search?query=alamo&page=2"/>
    <meta property="og:url" content="http://localhost:8080/search?query=alamo&page=2"/>
```

Reply:
If we really wanted to make the test explicit (rather than in a roundabout way via a full snapshot), we could probably drill into the `HTML` node we're getting back and inspect the elements directly but it's probably not worth the effort.

Reply:
Sorry for the spam, I had another thought. We could also write this as follows, without any snapshotting:

```swift
    func test_SearchShow_Model_canonicalURLAllowList() async throws {
        let req = Vapor.Request(application: app,
                                url: "search?query=alamo&page=2&utm_campaign=test&utm_source=email",
                                on: app.eventLoopGroup.next())
        let html = try await SearchController.show(req: req).render(indentedBy: .spaces(2))
        XCTAssert(
            html.contains(#"<link rel="canonical" href="http://localhost:8080/search?query=alamo&page=2"/>"#),
            "was: \(html)"
        )
    }
```

This way it's very explicit what we're looking for. It has the downside that (unless we added another check to constrain the number of `link rel` snippets to 1) we're not asserting that there's a second one with the extra parameters.